### PR TITLE
frameworks: added dropout in torch nn modules along.

### PR DESCRIPTION
**Pull request by shubham3121**
# Pull Request

## Description

Added support for nn.Dropout layer.

Issue: #3497 

## Affected Dependencies
List any dependencies that are required for this change.

## Type of Change
Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation (non-breaking change which adds documentation)
- [ ] Improvement (non-breaking change that improves the performance or reliability of existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I did follow the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md)
- [x] I did follow the [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have added tests for my changes

## Additional Context
Please also include relevant motivation and context.

### CODE REVIEW

File: syft/frameworks/torch/nn/dropout.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+from torch.nn import Module
```
[the remark this answers is not in the document]
Comment (shubham3121) [reply syft/frameworks/torch/nn/dropout.py:1]: PySyft cannot hook into layers which are implemented in C++ and dropout layer uses the dropout function implemented in C++. Its similar to the conv2d or pooling layers implemented in Pysyft.

File: syft/frameworks/torch/nn/functional.py [hunk 2]
```diff
@@ -20,10 +20,10 @@ def dropout(input, p=0.5, training=True, inplace=False):
         binomial = torch.distributions.binomial.Binomial(probs=1 - p)
 
         # we must convert the normal tensor to fixed precision before multiplication
-        # Note that: Weights of a model are alwasy Float values
+        # Note that: Weights of a model are always Float values
         # Hence input will always be of type FixedPrecisionTensor > ...
         noise = (binomial.sample(input.shape).type(torch.FloatTensor) * (1.0 / (1.0 - p))).fix_prec(
-            **input.get_class_attributes(), no_wrap=True
+            no_wrap=True
```
[the remark this answers is not in the document]
Comment (shubham3121) [reply syft/frameworks/torch/nn/functional.py:26]: @LaRiffle The reason I removed it was that `input.get_class_attributes()` returned a dictionary `{"type": torch.float32}` and in turn is passed to syft.FixedPrecisionTensor as kwargs inside fix_prec function. But since FixedPrecisionTensor doesn't have any variable 'type' it throws error unexpected key argument.
